### ISSUE TITLE
When server sends too much logs, GUI stops responding usually for 10-20 minutes

### ISSUE THREAD

**Issue by vadim-a-yegorov** [Lodestone-Team/lodestone#289]
# Description

When server sends more than 3000 log strings in total, lodestone GUI stucks. Sometimes scrolling stucks for a few seconds, sometimes the whole GUI stops responding forever and requires closing and reopening.

# How to solve

- [ ] Debug (or implement) async/await mechanism, retrieving logs shouldn't freeze GUI

**Comment by CheatCod** [comment]
We never ran into a situation when we need to render that many logs before, so this flew under our radar.

Lodestone Core stores a fixed-size buffer of the last 1024 console logs, which is fetched when you refresh the page.

The temporary solution ATM is to refresh the page if you are using the web app, or restart if you are using the desktop client. I understand it's annoying and we thank you for using Lodestoen and reporting all the bugs that slipped through. Please bear with us as we work on a fix.

**Comment by vadim-a-yegorov** [comment]
> We never ran into a situation when we need to render that many logs before, so this flew under our radar.
> 
> Lodestone Core stores a fixed-size buffer of the last 1024 console logs, which is fetched when you refresh the page.
> 
> The temporary solution ATM is to refresh the page if you are using the web app, or restart if you are using the desktop client. I understand it's annoying and we thank you for using Lodestone and reporting all the bugs that slipped through. Please bear with us as we work on a fix.

When I'm trying to restart an app I run into 4 other issues which takes 3 minutes to login again. I understand that you're doing your best and I like your product but it needs a few small fixes to be perfect.

**Comment by vadim-a-yegorov** [comment]
> We never ran into a situation when we need to render that many logs before, so this flew under our radar.
> 
> Lodestone Core stores a fixed-size buffer of the last 1024 console logs, which is fetched when you refresh the page.
> 
> The temporary solution ATM is to refresh the page if you are using the web app, or restart if you are using the desktop client. I understand it's annoying and we thank you for using Lodestone and reporting all the bugs that slipped through. Please bear with us as we work on a fix.

I run commercial minecraft fabric server with 115 server mods, 9 cardboard plugins and 59 client mods. Lodestone probably wasn't tested for it.

**Comment by CheatCod** [comment]
You are correct, and it's on us that we didn't test our code as thoroughly as we could have.

I would like to take a minute to explain why you are running into so many issues we failed to catch, and how we plan to respond and improve for the future,

1. MacOS desktop client. No one on our dev team has consistent access to a Mac machine, thus we couldn't test it at all. In my defense, the installation guide clearly states "Lodestone Desktop for MacOs is highly experimental and untested. We won't be able to provide support if you decide to use it.", and the release page labels MacOS desktop client as "🚧 experimental and untested". This does not take away the fact that our software is broken and untested on MacOS. To address this, we plan to either disable Mac builds altogether until we can properly test it, or push the user harder to use our web app https://www.lodestone.cc/ instead.
2. Using the desktop client to connect to a remote core. Lodestone was designed to be noob friendly, so the setup experience for the desktop client was focused on "getting the user setup with a Minecraft server ASAP", thus we overlooked the use case for using a desktop client to connect to a remote core. This is completely on us, we will be testing more thoroughly before release.
3. Use of a HTTPS domain. Lodestone is designed to be noob friendly as I mentioned before, so we built in the feature for HTTPS connection but never bothered to set up a server behind said HTTPS domain and test it. This is again on us.
4. Use in a commercial setting. Lodestone's primary focus is the casual userbase, and any commercial users we got do not have a setup even close to yours, so we were never heavy on testing for the high-workload cases. It's also rather cumbersome to test for high-workload cases as we will need a BIG Minecraft server setup.

Again, thanks for digging up all the issues that we missed. We will be working through them as fast as we can.

In the meantime, let me know if https://www.lodestone.cc/ addresses some of your issues/makes them more bearable.

Please feel free to report any more defects/suggestions!

**Comment by vadim-a-yegorov** [comment]
I like this project because it's the only one server launcher that just works. It took me 1 minute to setup remote server and install local GUI. It may take a little more effort to make it production ready for servers of any size. After fixing all essential issues and making it production ready, project direction can be changed to add new cool features, such as automatic mod or modpack install from Modrinth, Curseforge, Technic, github, custom file or URL; check mods for update, display them as a list with names, icons and descriptions; enhanced smooth file explorer with more functionality and invisible delays; ssh/sftp/vnc/rdp shortcuts; embedded json/yaml/properties/toml editor; display more data on the server status tab like active players/afk players/entities loaded/chunks loaded/storage usage/TPS/average ping/100+ other indicators; JMX GUI; JVM profiling data that will help detect memory leaks like spark, grafana and prometheus. It's up to you what you think is good to implement and what is not, it's just my suggestion.

**Comment by vadim-a-yegorov** [comment]
This issue occurs both on:
* Electron (macOS Monterey 12.3.1 tested)
* Browser (Google Chrome Version 114.0.5735.198 (Official Build) (x86_64) tested)
